zypp/Package.cc

```diff
@@ -26,6 +27,61 @@ namespace zyppintern
 {
   using namespace zypp;
 
+  /** getVendorSupportOption return value */
+  using getVendorSupportOptionReturn = std::pair<VendorSupportOption,std::optional<std::set<std::string_view>>>;
+
+  /** Get VendorSupportOption and the PKGNAME if VendorSupportSuperseded */
+  getVendorSupportOptionReturn getVendorSupportOption( sat::Solvable solv_r )
+  {
+    static const IdString support_unsupported( "support_unsupported" );
+    static const IdString support_acc( "support_acc" );
+    static const IdString support_l1( "support_l1" );
+    static const IdString support_l2( "support_l2" );
+    static const IdString support_l3( "support_l3" );
+    static const IdString support_superseded( "support_superseded_by(" ); // followed by "PKGNAME)"
+
+    getVendorSupportOptionReturn ret { VendorSupportUnknown, std::nullopt };
+
+    // max over all identical packages
+    for ( const auto & solv : sat::WhatProvides( (Capability(solv_r.ident().id())) ) )
+    {
+      if ( solv.edition() == solv_r.edition()
+        && solv.ident() == solv_r.ident()
+        && solv_r.identical( solv ) )
+      {
+        for ( PackageKeyword kw : Package::Keywords( sat::SolvAttr::keywords, solv ) )
+        {
+          switch ( ret.first )
+          {
+            case VendorSupportUnknown:
+              if ( kw == support_unsupported )	{ ret.first = VendorSupportUnsupported; break; }
+            case VendorSupportUnsupported:
+              if ( kw == support_acc )	{ ret.first = VendorSupportACC; break; }
+            case VendorSupportACC:
+              if ( kw == support_l1 )	{ ret.first = VendorSupportLevel1; break; }
+            case VendorSupportLevel1:
+              if ( kw == support_l2 )	{ ret.first = VendorSupportLevel2; break; }
+            case VendorSupportLevel2:
+              if ( kw == support_l3 )	{ ret.first = VendorSupportLevel3; break; }
+            case VendorSupportLevel3:
+              // VendorSupportSuperseded is special as the kw may also contain a '(PKGNAME)'.
+              // even more: multiple
+              if ( strv::hasPrefix( kw.asStringView(), support_superseded.asStringView() ) && kw.c_str()[kw.size()-1] == ')' ) {
+                ret.first = VendorSupportSuperseded;
+                if ( not ret.second )
+                  ret.second = std::set<std::string_view>();
+                ret.second->insert( std::string_view( kw.c_str()+support_superseded.size(), kw.size()-support_superseded.size()-1 ) );
```

Comment:
Not sure if I get this right, but once we encountered suuperseded once this case won't be entered again because ret.first is set to VendorSupportSuperseded ... so in the next for loop iteration the code would end up in the next case branch , so the Set will always only have one entry even if we encounter the superseded_by tag again

Reply:
fixed